Counterexamples/Monic_nonRegular.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2023 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Mathlib.Data.Polynomial.Monic
+
+/-!
+# `Monic` does not necessarily imply `IsRegular` in a `Semiring` with no opposites
+
+This counterexample shows that the hypothesis `Ring R` cannot be changed to `Semiring R` in
+`Polynomial.Monic.isRegular`.
+
+The example is very simple.
+
+The coefficient Semiring is the Semiring `N` of the natural numbers `{0, 1, 2, 3}` where the
+standard addition and standard multiplication are truncated at `3`.
+
+The products `(X + 2) * (X + 2)` and `(X + 2) * (X + 3)` are equal to
+`X ^ 2 + 4 * X + 4` and `X ^ 2 + 5 * X + 6` respectively.
+
+By truncation, `4, 5, 6` all mean `3` in `N`.
+It follows that multiplication by `(X + 2)` is not injective.
+-/
+open Polynomial
+
+namespace nonRegular
```

Comment:
```suggestion
namespace Counterexample.NonRegular
```
since everything in this folder lives in `Counterexample`

Reply:
Of course!  I might have been the one who actually implemented the namespacing... :facepalm: